### PR TITLE
gwt 2.9.0

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/gwt.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Gwt < Formula
   desc "Google web toolkit"
   homepage "http://www.gwtproject.org/"
-  url "https://storage.googleapis.com/gwt-releases/gwt-2.8.2.zip"
```
Comment (chenrui333) [inline Formula/gwt.rb:4]: In the project page, it still shows 2.8.2 though.

Comment (chenrui333) [reply Formula/gwt.rb:4]: cc @Bo98

**Comment by samford** [comment]
Hm, this situation is kind of a mixed bag:

* The 2.9.0 release is on a separate branch (`releases/2.9.0`) in [the upstream repository](https://gwt.googlesource.com/gwt.git) and the `2.9.0` tag points to that branch. The other version tags (e.g., `2.9.0-RC1`, `2.8.2`, etc.) point to commits in the master branch. The commits in the `releases/2.9.0` branch matches `master` except that the two commits between the `2.9.0-RC1` tag and `2.9.0` tag have different commit hashes (same title and author, though). It's weird that they didn't just create the `2.9.0` tag on master like they've been doing.
* The upstream website doesn't list 2.9.0 yet. Looking at [the repository for their website](https://github.com/gwtproject/gwt-site), it seems like they have to manually update the website for each release. The release before 2.9.0 was around 1.5 years prior, so it's hard to say whether they're waiting for something or if they just haven't gotten around to it yet.
* [Other software repositories haven't upgraded to 2.9.0 yet](https://repology.org/project/gwt/versions).

I just wanted to post some additional information and you can take from this what you will. It may be that it's appropriate to wait for the release to hit the website. Since the last release was 1.5 years ago, I can't tell what their typical process is like.

File: Formula/gwt.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/gwt.rb:4]: made a release notes update request, https://github.com/gwtproject/gwt-site/issues/298

**Comment by tbroyer** [comment]
> Hm, this situation is kind of a mixed bag:
> 
> * The 2.9.0 release is on a separate branch (`releases/2.9.0`) in [the upstream repository](https://gwt.googlesource.com/gwt.git) and the `2.9.0` tag points to that branch. The other version tags (e.g., `2.9.0-RC1`, `2.8.2`, etc.) point to commits in the master branch. The commits in the `releases/2.9.0` branch matches `master` except that the two commits between the `2.9.0-RC1` tag and `2.9.0` tag have different commit hashes (same title and author, though). It's weird that they didn't just create the `2.9.0` tag on master like they've been doing.

GWT uses [Trunk-Based Development](https://trunkbaseddevelopment.com/) with [branches for releases](https://trunkbaseddevelopment.com/branch-for-release/), created as late as possible and only when necessary (which is why some releases are tagged on the master branch). Since 2.9.0-RC1, a couple commits have been made that are breaking changes, then a couple commits fixing issues with the RC1, so a branch has been created, those commits cherry-picked, and the 2.9.0 tagged.

> * The upstream website doesn't list 2.9.0 yet. Looking at [the repository for their website](https://github.com/gwtproject/gwt-site), it seems like they have to manually update the website for each release. The release before 2.9.0 was around 1.5 years prior, so it's hard to say whether they're waiting for something or if they just haven't gotten around to it yet.
> * [Other software repositories haven't upgraded to 2.9.0 yet](https://repology.org/project/gwt/versions).

2.9.0 is actually still being tested and hasn't been released yet (should be very soon though).

> I just wanted to post some additional information and you can take from this what you will. It may be that it's appropriate to wait for the release to hit the website. Since the last release was 1.5 years ago, I can't tell what their typical process is like.

Yes, please wait for the announcement.

**Comment by samford** [comment]
Thank you, @tbroyer! That information's very good to know going forward. We'll be sure to wait for a release announcement before updating to a new version.

**Comment by niloc132** [comment]
Release announcement went out on the mailing list last night, that 2.9.0 link is official! However, we're experiencing issues of updating gwtproject.org, so it isnt visible there yet.

Here's the announcement and release notes: https://groups.google.com/d/msg/google-web-toolkit/kXWfvW9X0pI/-lxHncJnAAAJ

**Comment by chenrui333** [comment]
> Release announcement went out on the mailing list last night, that 2.9.0 link is official! However, we're experiencing issues of updating gwtproject.org, so it isnt visible there yet.
> 
> Here's the announcement and release notes: https://groups.google.com/d/msg/google-web-toolkit/kXWfvW9X0pI/-lxHncJnAAAJ

wonderful, do you think we can go with the merge now?

**Comment by niloc132** [comment]
The release is official, and posted on the site! http://www.gwtproject.org/release-notes.html#Release_Notes_2_9_0 Please feel free to merge.

**Comment by chenrui333** [comment]
Thanks @niloc132 for the update. Merging now!! 🎉

**Pull request merged**